### PR TITLE
Add example notebooks that use application objects

### PR DESCRIPTION
**Pull request by msschwartz21**
## What
* Add example notebooks that use application objects for cytoplasmic and nuclear segmentation

## Why
* Demonstrate the utility of application objects and serves as a starting point for new users

Currently I am building this branch on [RTD](https://deepcell.readthedocs.io/en/notebooks/index.html) so that we can check how the notebooks render for documentation. I will deactivate the branch on RTD after we are done with this PR.

### CODE REVIEW

**Comment by msschwartz21** [comment]
@willgraf Can we exclude notebooks from testing coverage? It's not really possible to test them so they are always going to hurt our coverage.

**Comment by willgraf** [comment]
> @willgraf Can we exclude notebooks from testing coverage? It's not really possible to test them so they are always going to hurt our coverage.

I believe that the coverage report is inside the `deepcell` module.

**Comment by willgraf** [comment]
What are the gif files? It seems like they are new files but are not used anywhere.

Additionally, why add the `.h5` and `.trk` files to `.gitignore`? Maybe we should ignore them for all the folders, rather than just inside `/docs`.

**Comment by willgraf** [comment]
Finally, do the notebooks belong inside `/docs`? Why not inside`/scripts`? I would think `/scripts` would be the right place for them, as they would go with all the other example scripts. I'm not sure RTD needs so many different notebooks.

**Comment by msschwartz21** [comment]
> What are the gif files? It seems like they are new files but are not used anywhere.

The gif files `tracks.gif` and `labeled.gif` are generated and then embedded in the nuclear segmentation and tracking notebooks. RTD uses them when converting the notebook to html.

> Additionally, why add the .h5 and .trk files to .gitignore? Maybe we should ignore them for all the folders, rather than just inside /docs.

I added them because running the example notebooks downloads h5 and trks files. I didn't want to risk them getting committed. I'll extend the ignore to all folders if there are no objections.

> Finally, do the notebooks belong inside /docs? Why not inside/scripts? I would think /scripts would be the right place for them, as they would go with all the other example scripts. I'm not sure RTD needs so many different notebooks.

I was following the previously discussed vision of having a few well documented notebooks included in our documentation. The idea being that the bulk of notebooks in scripts are not well documented and are too numerous to easily maintain high-quality documentation. 

Practically they are insides the docs folder because sphinx cannot "see" files outside the docs with the exception `README.md`. One alternative that I investigated a while back would be to copy essential notebooks for documentation from scripts into docs at the start of the sphinx build. I think that way we could avoid having them under git tracking in the docs folder. 

However, I think the quality of our example notebooks will be better if we commit to maintaining a select subset as opposed to all notebooks in scripts.

**Comment by willgraf** [comment]
>I was following the previously discussed vision of having a few well documented notebooks included in our documentation. The idea being that the bulk of notebooks in scripts are not well documented and are too numerous to easily maintain high-quality documentation.

Makes sense. The only issue I see is that these notebooks are only visible from RTD, and users cannot interact with them via GitHub or a docker container. I was under the impression we were just going to clean up the notebooks inside `/scripts` which has been the typical entry-point for new users.

**Comment by msschwartz21** [comment]
> The only issue I see is that these notebooks are only visible from RTD, and users cannot interact with them via GitHub or a docker container. 

On RTD, we add a banner that directs people to go to Github to view and/or download the notebook. I'm working with Ed to try to "test" the notebook and see if it is sufficient to get him started.
Screen Shot 2020-04-14 at 11 10 36 AM

I'm not sure what to do about the scripts folder. Maybe that is worth having a group conversation about?

**Comment by willgraf** [comment]
I think the notebook looks good, but I might be on a different page regarding the best way to interact with it. I'll defer to @vanvalen for the review and what would be most helpful to users.

**Comment by msschwartz21** [comment]
After talking to Dave, I am going to add the current example notebooks as is with a note that we may need to improve them for novice users in the future. Currently the nuclear and cytoplasm example notebooks are not included in the documentation and instead live in the `/scripts` folder.

File: .gitignore [hunk 1]
```diff
@@ -119,3 +119,8 @@ venv.bak/
 .Trashes
 ehthumbs.db
 Thumbs.db
+
+# Documentation notebook byproducts
+*.h5
+docs/source/nucleartracks/*
```
[the remark this answers is not in the document]
Comment (msschwartz21) [reply .gitignore]: I removed the docs/source line but kept the data files since I think that is generally a good precaution.

**Comment by msschwartz21** [comment]
I removed most of the extraneous changes that you noticed with the exception of the `nbsphinx_prolog` change to `conf.py`. That produces a header on any notebook that we render in our documentation and directs the user back to github to download the original notebook.

**Pull request merged**